core/src/main/java/io/aiven/klaw/service/BaseOverviewService.java

```diff
@@ -78,9 +84,132 @@ protected List<AclInfo> getAclInfoList(
     }
 
     topicOverview.setTopicInfoList(topicInfoList);
+    sortAclInfo(aclInfo, groupBy);
     return aclInfo;
   }
 
+  private void sortAclInfo(List<AclInfo> aclInfo, AclGroupBy groupBy) {
+    // merge reduce
+    if (groupBy.equals(AclGroupBy.TEAM)) {
+      mergeAclInfo(aclInfo);
+      Collections.sort(aclInfo, Comparator.comparing(AclInfo::getTeamname));
+    }
+  }
+
+  private void mergeAclInfo(List<AclInfo> aclInfo) {
+    List<AclInfo> organisedList = new ArrayList<>();
+
+    while (aclInfo.size() > 0) {
+      List<AclInfo> team = new ArrayList<>();
+      List<AclInfo> unprocessable = new ArrayList<>();
+      String teamname = aclInfo.get(0).getTeamname();
+      for (AclInfo info : aclInfo) {
```

Comment:
using multiple groupingBy (team, env and aclType) and merge the final lists based on key ((ip, principal)) could be better ?